main/load-tx-rlp.zkasm

```diff
@@ -290,30 +290,22 @@ vREADTx:
 ;;;;;;;;;
 ;; update bytes parsed
         $ => A                          :MLOAD(batchL2DataParsed)
-        A + C => A                      :MSTORE(batchL2DataParsed)
+        A + C                           :MSTORE(batchL2DataParsed)
 ;; increase number of transaction to process
         $ => A                          :MLOAD(pendingTxs)
-        A + 1 => A                      :MSTORE(pendingTxs)
+        A + 1                           :MSTORE(pendingTxs)
 ;; compute signature
         $ => A                          :HASHKDIGEST(E)
         A                               :MSTORE(txHash), JMP(txLoopRLP)
 
 ;;;;;;;;;
 ;; E - Handler error RLP fields
 ;;;;;;;;;
-handleOOCBatRLP:
-        $${eventLog(onError, OOCB)}      :JMP(handleOOCatRLP)
-handleOOCKatRLP:
-        $${eventLog(onError, OOCK)}      :JMP(handleOOCatRLP)
-handleOOCSatRLP:
-        $${eventLog(onError, OOCS)}      :JMP(handleOOCatRLP)
-handleOOCatRLP:
-        $ => SR                          :MLOAD(batchSR)
-
 invalidTxRLP:
 ;; Append all missing 'batchL2Data' to 'batchDataHash' bytes
         $ => B                          :MLOAD(batchL2DataLength)
         $ => C                          :MLOAD(batchHashPos)
+        $${p = C}
```

Comment:
Why is this needed? Seems like it can change behaviour (of `getTxs()`?) significantly

Reply:
`p` is the pointer to the read transactions bytes via `getTxs()`.

While reading RLP, we add bytes to the `batchHash` and perform some check afterwards, but the bytes are already added. So, if anything goes wrong we jump to `invalidTxRLP` to load the remaining bytes. The remaining bytes where added starting at `p`, where the `p` was set during the RLP parsing. This was always true before, but if we perform the check before the bytes are added to the `batchHash`, then it will not match the `batchHash`.ç

The generic way to add the missing `batchHash` bytes was to simply set the pointer `p` to starting byte to add, which is stored in `batchHashPos`

Reply:
> So as I understand this error processing is now called from more places (`utils` generating out-of-counters errors), including from the points where `p` is not updated yet, so it is initialized explicitly.

exactly!

Reply:
> > The remaining bytes where added starting at `p`, where the `p` was set during the RLP parsing.
> 
> But for old cases, where this error processing is called after `p` is already set, wouldn't this code here add more bytes than needed?

for old cases, new code will just only set the `p` where it was already read, so it will just only rewrite the `p` with same value